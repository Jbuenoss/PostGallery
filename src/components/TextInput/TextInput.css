.input-search{
    padding: 10px;
    font-size: 1.2em;
    width: 100%;
}