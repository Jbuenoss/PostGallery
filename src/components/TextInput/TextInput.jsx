import './TextInput.css'
export const TextInput = ({ handleChange, searchValue }) => (
    <input type="search"
        onChange={handleChange}
        value={searchValue}
        className="input-search" />
);
