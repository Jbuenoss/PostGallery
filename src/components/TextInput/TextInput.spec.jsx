import userEvent from '@testing-library/user-event';
import { TextInput } from './TextInput.jsx';
import { render, screen } from '@testing-library/react';

describe('<Posts />', () => {
    it('should have a value of searchValue', () => {
        const fn = jest.fn();
        render(<TextInput searchValue="Testing" handleChange={fn}/>);
        
        const input = screen.getByPlaceholderText(/type your search/i);
        expect(input.value).toBe('Testing');
    });

    it('should call handleChange function on each key pressed', async () => {
        const fn = jest.fn();
        render(<TextInput handleChange={fn}/>);
        
        const input = screen.getByPlaceholderText(/type your search/i);
        const value = 'the value';
        await userEvent.type(input, value);
        
        expect(input.value).toBe(value);
        expect(fn).toHaveBeenCalledTimes(value.length)
    });

    it('should match snapshot', () => {
        const fn = jest.fn();
        const {container} = render(<TextInput handleChange={fn} />);

        expect(container.firstChild).toMatchSnapshot();
    });
});