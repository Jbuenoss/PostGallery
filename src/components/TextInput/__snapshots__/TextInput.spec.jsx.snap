// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Posts /> should match snapshot 1`] = `
<input
  class="input-search"
  placeholder="Type your search"
  type="search"
  value=""
/>
`;
